### ISSUE TITLE
SiteImprove revealed dspace code accessibility Issues

### ISSUE THREAD

**Issue by emj73** [DSpace/dspace-angular#2055]
We ran a SiteImprove audit on our dev dpsace 7 and the following issues were revealed that target dspace out of the box code: [https://cornell.box.com/s/xo17pd4sfcrd314dul32w70lbdqsg8i8]

please let me know if you are unable to access that link. 

thanks for considering :)

**Comment by emj73** [comment]
for ease, here is the info from the doc copied: 

Headings are not structured 

- Headings are out of order. 

https://demo7.dspace.org/items/a1283d9b-8702-46ee-acb8-1d19302349b2 

Item title is H2. Next heading is H5 (Files, Date, Author, etc.). 

 

Element IDs are not unique 

- The element search-container-navbar appears twice in the header code. 

https://github.com/DSpace/dspace-angular/tree/main/src/app/search-navbar  

https://demo7.dspace.org/items/a1283d9b-8702-46ee-acb8-1d19302349b2 

 

State or property not supported 

-  Aria-disabled attribute is not supported on an <a> tag in this context. This is on disabled links in pagination. 

https://demo7.dspace.org/search?query=history 

<a aria-label="Previous" href="" class="page-link" tabindex="-1" aria-disabled="true">...</a> 

 

Button missing a text alternative 

- There is a button that does not actually display in this view, but the HTML is still there. It comes before the “Show more” button in the code. If you run the WAVE plugin on this page it shows four instances of empty buttons. 

https://demo7.dspace.org/community-list 

Example: 

<button type="button" cdktreenodetoggle="" class="btn btn-default"><span aria-hidden="true" class="fa fa-chevron-right invisible"></span></button>

**Comment by emj73** [comment]
some of the above comment cut off: 

Button missing a text alternative 

- There is a button that does not actually display in this view, but the HTML is still there. It comes before the “Show more” button in the code. If you run the WAVE plugin on this page it shows four instances of empty buttons. 

https://demo7.dspace.org/community-list 

Example: 

<button type="button" cdktreenodetoggle="" class="btn btn-default"><span aria-hidden="true" class="fa fa-chevron-right invisible"></span></button>

**Comment by tdonohue** [comment]
At a quick glance, I *believe* many of these issues overlap with existing, open `accessibility` tickets: https://github.com/DSpace/dspace-angular/issues?q=is%3Aissue+is%3Aopen+accessibility+label%3Aaccessibility

We are aware there are a number of buttons with missing text, non-unique elements, etc.  We've slowly been fixing those in page-specific accessibility tickets.
